compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/WindowExceptionHandlerFactory.desktop.kt

```diff
@@ -0,0 +1,37 @@
+package androidx.compose.ui.awt
+
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.ExperimentalComposeUiApi
+import java.awt.Window
+import java.awt.event.WindowEvent
+import javax.swing.JFrame
+import javax.swing.JOptionPane
+import javax.swing.SwingUtilities
+
+/**
+ * Default [WindowExceptionHandlerFactory], which will show standard error dialog, and close the window after that
+ */
+@ExperimentalComposeUiApi
+object DefaultWindowExceptionHandleFactory : WindowExceptionHandlerFactory {
+    override fun exceptionHandler(window: Window) = Thread.UncaughtExceptionHandler { _, throwable ->
+        SwingUtilities.invokeLater {
+            JOptionPane.showMessageDialog(window, "An error has occurred", "Error", JOptionPane.ERROR_MESSAGE)
+            window.dispatchEvent(WindowEvent(window, WindowEvent.WINDOW_CLOSING))
```

Comment:
Closing window may look strange, i.e. imagine window opened a popup and exception happened in main window. Maybe better close application as the whole, although both approaches rather nasty if some unsaved state exists.

Reply:
Window usually self-contained, and manipulate state within the window.

But yeah, sometimes exception can happen during writing to the global state, which is dangerous to leave it broken.

But probably it is rare, and usually it is a mistake of the developer. In majority of cases, it is better to close the window than the application, so user won't lose the unsaved state.

P.S.
closing only the popup probably is much less desirable, and is more complex task